src/kronecker.jl

```diff
@@ -0,0 +1,258 @@
+            promote_type(T, eltype(A)) == T || throw(InexactError())
+        end
+        return new{T,As}(maps)
+    end
+end
+
+KroneckerSumMap{T}(maps::As) where {T, As<:Tuple{LinearMap,LinearMap}} = KroneckerSumMap{T, As}(maps)
+
+"""
+    kronsum(A::LinearMap, B::LinearMap)
+    kronsum(A, B, Cs...)
+
+Construct a `KroneckerSumMap <: LinearMap` object, a (lazy) representation of the
+Kronecker sum `A⊕B = kron(A, Ib) + kron(Ia, B)` of two square `LinearMap`s. Here,
+`Ia` and `Ib` are identity operators of the size of `A` and `B`, respectively.
+Arguments of type `AbstractMatrix` are promoted to `LinearMap`s as long as there
+is a `LinearMap` object among the first 8 arguments.
+
+# Examples
+```jldoctest; setup=(using LinearAlgebra, SparseArrays, LinearMaps)
+julia> J = LinearMap(I, 2) # 2×2 identity map
+LinearMaps.UniformScalingMap{Bool}(true, 2)
+
+julia> E = spdiagm(-1 => trues(1)); D = LinearMap(E + E' - 2I);
+
+julia> Δ₁ = kron(D, J) + kron(J, D); # discrete 2D-Laplace operator
+
+julia> Δ₂ = LinearMaps.kronsum(D, D); # same operator as Kronecker sum
+
+julia> Matrix(Δ₁) == Matrix(Δ₂)
+true
+```
+"""
+kronsum(A::LinearMap{TA}, B::LinearMap{TB}) where {TA,TB} = KroneckerSumMap{promote_type(TA,TB)}((A, B))
+kronsum(A::LinearMap, B::LinearMap, C::LinearMap, Ds::LinearMap...) = kronsum(A, kronsum(B, C, Ds...))
+# promote AbstractMatrix arguments to LinearMaps, then take LinearMap-Kronecker sum
+for k in 1:8 # is 8 sufficient?
+    Is = ntuple(n->:($(Symbol(:A,n))::AbstractMatrix), Val(k-1))
+    # yields (:A1, :A2, :A3, ..., :A(k-1))
+    L = :($(Symbol(:A,k))::LinearMap)
+    # yields :Ak
+    mapargs = ntuple(n -> :(LinearMap($(Symbol(:A,n)))), Val(k-1))
+    # yields (:LinearMap(A1), :LinearMap(A2), ..., :LinearMap(A(k-1)))
+
+    @eval kronsum($(Is...), $L, As::Union{LinearMap,AbstractMatrix}...) =
+        kronsum($(mapargs...), $(Symbol(:A,k)), promote_to_lmaps(As...)...)
+end
+
+Base.size(A::KroneckerSumMap, i) = prod(size.(A.maps, i))
+Base.size(A::KroneckerSumMap) = (size(A, 1), size(A, 2))
+
+LinearAlgebra.issymmetric(A::KroneckerSumMap) = all(issymmetric, A.maps)
+LinearAlgebra.ishermitian(A::KroneckerSumMap{<:Real}) = all(issymmetric, A.maps)
+LinearAlgebra.ishermitian(A::KroneckerSumMap) = all(ishermitian, A.maps)
+
+LinearAlgebra.adjoint(A::KroneckerSumMap{T}) where {T} = KroneckerSumMap{T}(map(adjoint, A.maps))
+LinearAlgebra.transpose(A::KroneckerSumMap{T}) where {T} = KroneckerSumMap{T}(map(transpose, A.maps))
+
+Base.:(==)(A::KroneckerSumMap, B::KroneckerSumMap) = (eltype(A) == eltype(B) && A.maps == B.maps)
+
+function LinearMaps.A_mul_B!(y::AbstractVector, L::KroneckerSumMap, x::AbstractVector)
+    A, B = L.maps
+    ma, na = size(A)
+    mb, nb = size(B)
+    (length(y) == size(L, 1) && length(x) == size(L, 2)) || throw(DimensionMismatch("kronecker product"))
+    X = reshape(x, (nb, na))
+    Y = reshape(y, (nb, na))
+    mul!(Y, X, convert(AbstractMatrix, transpose(A)))
```

Comment:
How about `transpose(convert(AbstractMatrix, A))`, to make use of the methods to multiply with transpose/adjoint matrices in Base. I was thinking about the case where `A` is a `MatrixMap`.

However, I just checked. There is no `convert(AbstractMatrix, ::LinearMap)`, only `convert(Matrix, ::LinearMap)`. So I assume this line of code is not  Furthermore, `convert(Matrix, ::MatrixMap)` does actually call the generic method instead of just returning the matrix it wraps. 

Never mind, I just noticed that you also fixed all of this in this PR :-).

Reply:
The conversion rules are most likely not complete. They cover what we used to have and matrix conversion that we need anyway in the Kronecker multiplication. I just thought that with block maps and 5-arg mul!, there is additional use of our package with only matrices, for which conversion doesn't necessarily require multiplying with tons of canonical vectors. So I collected whatever came to my mind would be cheaply obtained directly from matrices. This could be extended as required.